tests/telemetry/test_writer.py

```diff
@@ -360,6 +360,7 @@ def test_add_integration_disabled_writer(telemetry_writer, test_agent_session):
     assert len(test_agent_session.get_requests()) == 0
 
 
+@flaky(until=1704067200)
 @pytest.mark.parametrize("mock_status", [300, 400, 401, 403, 500])
 def test_send_failing_request(mock_status, telemetry_writer):
```

Comment:
This [failure](https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/49982/workflows/3cc306aa-bd2a-4c4c-aed5-44f506b97e99/jobs/3201301) is really peculiar. For some reason httpretty failed to mock the endpoint and a real request was sent. I have never seen this failure before and I don't think we made recent changes to how we send telemetry is sent or the version of httpretty that is installed. 

Can we add an option to `tests.utils.flaky()` to mark tests as `xfail`? It would be nice if we could still run some flaky tests but not block ci if it fails.